frame/babe/src/lib.rs

```diff
@@ -483,8 +494,16 @@ impl<T: Config> Module<T> {
 		};
 		Self::deposit_consensus(ConsensusLog::NextEpochData(next_epoch));
 
-		if let Some(next_config) = NextEpochConfig::take() {
-			Self::deposit_consensus(ConsensusLog::NextConfigData(next_config));
+		if let Some(pending_epoch_config_change) = PendingEpochConfigChange::take() {
+			if let Some(next_config) = NextEpochConfig::get() {
+				EpochConfig::put(next_config);
+			}
```

Comment:
So when there is no `PendingEpochConfigChange`, we don't apply the next epoch config?

Reply:
No, don't think so.

Reply:
So this is a bug?

Reply:
@bkchr is correct here. 

```
if let Some(next_config) = NextEpochConfig::get() {
	EpochConfig::put(next_config);
}
```

This code should exist outside the outer `if`, it should happen unconditionally regardless of whether there's a planned change or not.